contrib/check_subp_boxes.py

```diff
@@ -14,63 +14,93 @@
 """
 
 import argparse
+from os import path as P
+
 import libadalang as lal
 
+from langkit.diagnostics import Location, print_error
 
-parser = argparse.ArgumentParser(description=__doc__)
-parser.add_argument('files', help='The files to analyze',
-                    type=str, nargs='+', metavar='F')
-parser.add_argument('--fix', action='store_true', help='Fix the code')
 
 content = []
 
 
 def check_unsync_box(subp_body):
     """
-    Check if subp_body has an unsynchronized box. If it does, returns the first
-    token of the box. If it doesn't return None.
+    Check the box of ``subp_body``:
+
+    - If it has none, return the first token of the subprogram.
+    - If it has one but it is malformed, return the first token of the box.
+    - If it has one and it's good, or if it is a toplevel subprogram and
+      doesn't need a box, return None.
 
     :type subp_body: lal.SubpBody
     :rtype: lal.Token|None
     """
+
+    def prev(tok):
+        while (tok := tok.previous) is not None and tok.kind == 'Whitespace':
+            continue
+        return tok
+
     ss = subp_body.f_subp_spec
 
     # Library unit subprogram with composed name
-    if type(ss.f_subp_name) is not lal.Identifier:
+    if type(ss.f_subp_name.f_name) is not lal.Identifier:
         return
 
-    subp_name = ss.f_subp_name.f_tok.text
-    t = subp_body.token_start.previous
+    subp_name = ss.f_subp_name.f_name.text
+    t = prev(subp_body.token_start)
 
     # Toplevel subprogram, no previous token
     if not t:
         return
 
-    t2 = t.previous
-    t3 = t2.previous
+    t2 = prev(t)
+    t3 = prev(t2)
     ts = [t, t2, t3]
+
     if all(t.kind == 'Comment' for t in ts) and all(
         c == '-' for c in t.text + t3.text
     ):
-        if t2.text != "-- {} --".format(subp_name):
+        if t2.text != f"-- {subp_name} --":
             return t3
-
-
-def main(args):
-    c = lal.AnalysisContext('utf-8')
-
-    for f in args.files:
+        else:
+            return None
+    else:
+        return subp_body.token_start
+
+
+class CheckSubpBoxes(lal.App):
+    def main(self):
+        # Sort unit by filename to have a deterministic processing order
+        for _, unit in sorted(self.units.items()):
+            self.process_unit(unit)
```

Comment:
IIUC the corresponding Langkit commit, this `main` overloading is redundant: could you remove it?